kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/history/presenter/OrderHistoryPresenter.java

```diff
@@ -160,8 +159,13 @@ private void showCouponDialog(Order order) {
     private Coupon deserializeCoupon(Order order) {
         try {
             CouponInfo couponInfo = gson.fromJson(order.getContent(), CouponInfo.class);
-            CouponCode couponCode = gson.fromJson(order.getResult().toString(), CouponCode.class);
-            return new Coupon(couponInfo, couponCode);
+            CouponCodeResult couponCodeResult = (CouponCodeResult)order.getResult();
```

Comment:
check that order.getResult() is CouponCodeResult

Reply:
don't need to, if it's not it will catch and return null, because it must be with a result.